src/api/sso/src/middleware.js

```diff
@@ -85,62 +85,88 @@ module.exports.captureAuthDetailsOnSession = function captureAuthDetailsOnSessio
   };
 };
 
-// Forward a request to create a new user to the Users service.
 module.exports.createTelescopeUser = function createTelescopeUser() {
   return async (req, res, next) => {
-    try {
-      const id = getUserId(req);
-      const response = await fetch(`${USERS_URL}/${id}`, {
-        method: 'POST',
-        headers: {
-          // re-use the user's authorization header and token
-          Authorization: req.get('Authorization'),
-          'Content-Type': 'application/json',
-        },
-        body: JSON.stringify(req.body),
-      });
-
-      if (response.status !== 201) {
-        logger.warn(
-          { status: response.status, id, data: req.body },
-          'unable to create user with Users service'
-        );
-        next(createError(response.status, 'unable to create user'));
-        return;
-      }
+    const id = getUserId(req);
+    const {
+      firstName,
+      lastName,
+      email,
+      displayName,
+      feeds,
+      github: { username: githubUserName, avatarUrl: githubAvatarUrl },
+    } = req.body;
 
-      next();
-    } catch (err) {
-      logger.error({ err }, 'error creating Telescope user');
-      next(createError(500, 'unable to create Telescope user'));
+    console.log('creator');
+    const { error: userError } = await supabase.from('telescope_profiles').insert(
+      {
+        id,
+        email,
+        first_name: firstName,
+        last_name: lastName,
+        display_name: displayName,
+        github_username: githubUserName,
+        github_avatar_url: githubAvatarUrl,
+      },
+      { returning: 'minimal' }
+    );
+
+    if (userError) {
+      logger.warn(
+        { code: userError.code, details: userError.details, id, data: req.body },
+        userError.message
+      );
+      next(createError(400, 'unable to create user'));
+      return;
+    }
+
+    const { error: feedError } = await supabase.from('feeds').insert(
```

Comment:
Can we run these two inserts in parallel vs `awaiting` both like this in sequence?

Reply:
People are requesting this feature https://github.com/supabase/supabase/discussions/526